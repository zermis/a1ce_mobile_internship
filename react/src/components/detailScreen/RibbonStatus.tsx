import React from 'react';
import { Text, StyleSheet, View } from 'react-native';

interface Props {
    status: number;
}

// This JSX.Element will be used in detail screen later on
const RibbonStatus = (props: Props): JSX.Element => {
    return (
        <View
            style={[
                styles.ribbon,
                props.status === 3 // Change later on after we refactor the code
                    ? { backgroundColor: 'orange' }
                    : { backgroundColor: 'green' }
            ]}
        >
            <Text style={styles.ribbonText}>{props.status}</Text>
        </View>
    );
};

const styles = StyleSheet.create({
    ribbon: {
        width: '25%',
        alignItems: 'center',
        justifyContent: 'center',
        borderRadius: 15,
        height: 24
    },
    ribbonText: {
        color: '#FFF',
        fontSize: 11
    }
});

export default RibbonStatus;
