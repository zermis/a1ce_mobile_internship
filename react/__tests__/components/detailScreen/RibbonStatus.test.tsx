import RibbonStatus from '../../../src/components/detailScreen/RibbonStatus';
import renderer from 'react-test-renderer';

it('renders ribbon correctly', () => {
    const tree = renderer.create(<RibbonStatus status={0} />).toJSON();
    expect(tree).toMatchSnapshot();
});
