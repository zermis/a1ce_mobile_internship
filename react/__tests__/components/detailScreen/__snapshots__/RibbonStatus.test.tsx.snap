// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders ribbon correctly 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "borderRadius": 15,
        "height": 24,
        "justifyContent": "center",
        "width": "25%",
      },
      Object {
        "backgroundColor": "green",
      },
    ]
  }
>
  <Text
    style={
      Object {
        "color": "#FFF",
        "fontSize": 11,
      }
    }
  >
    0
  </Text>
</View>
`;
